test/lib/jdk/test/lib/hotspot/ir_framework/TestFramework.java

```diff
@@ -0,0 +1,1226 @@
+     *     <li><p>If a class is used by the test class that does not specify any compile command annotations, you do not
+     *            need to include it with this method. If no helper class specifies any compile commands, you do
+     *            not need to call this method at all.</li>
+     * </ul>
+     *
+     * <p>
+     * The testing can be started by invoking {@link #start()}
+     *
+     * @param helperClasses helper classes containing compile command annotations ({@link ForceCompile},
+     *                      {@link DontCompile}, {@link ForceInline}, {@link DontInline}) to be applied
+     *                      while testing {@code testClass} (also see description of {@link TestFramework}).
+     * @return the same framework instance.
+     */
+    public TestFramework addHelperClasses(Class<?>... helperClasses) {
+        TestRun.check(helperClasses != null && Arrays.stream(helperClasses).noneMatch(Objects::isNull), "A Helper class cannot be null");
+        if (this.helperClasses == null) {
+            this.helperClasses = new ArrayList<>();
+        }
+
+        for (Class<?> helperClass : helperClasses) {
+            TestRun.check(!this.helperClasses.contains(helperClass), "Cannot add the same class twice: " + helperClass);
+            this.helperClasses.add(helperClass);
+        }
+        return this;
+    }
+
+    /**
+     * Add scenarios to be used for the test VM. A test VM is called for each scenario in {@code scenarios} by using the
+     * specified VM flags in the scenario. The scenario flags override any flags set by {@link #addFlags(String...)}
+     * and thus also override any VM or Javaoptions set by JTreg by default.<p>
+     * Use {@code -DPreferCommandLineFlags=true} if you want to prefer the VM and Javaoptions over the scenario flags.
+     *
+     * <p>
+     * The testing can be started by invoking {@link #start()}
+     *
+     * @param scenarios scenarios which specify specific flags for the test VM.
+     * @return the same framework instance.
+     */
+    public TestFramework addScenarios(Scenario... scenarios) {
+        TestRun.check(scenarios != null && Arrays.stream(scenarios).noneMatch(Objects::isNull), "A scenario cannot be null");
+        if (this.scenarios == null) {
+            this.scenarios = new ArrayList<>(Arrays.asList(scenarios));
+        } else {
+            this.scenarios.addAll(Arrays.asList(scenarios));
+        }
+        return this;
+    }
+
+    /**
+     * Start the testing of the implicitly (by {@link #TestFramework()}) or explicitly (by {@link #TestFramework(Class)})
+     * set test class.
+     */
+    public void start() {
+        installWhiteBox();
+        maybeDisableIRVerificationCompletely();
+
+        if (scenarios == null) {
+            try {
+                start(null);
+            } catch (TestVMException e) {
+                System.err.println("\n" + e.getExceptionInfo());
```

Comment:
you shouldn't use "\n", as it might not be the right line-separator. you can either do:
```suggestion
                System.err.println();
                System.err.println(e.getExceptionInfo());
```
or
```suggestion
                System.err.printf("%n%s%n", e.getExceptionInfo());
```

Reply:
sure you case use `System.lineSeparator()`, it's just a matter of personal choice/style

Reply:
:+1: I fixed them.